waku/waku_store_legacy/client.nim

```diff
@@ -0,0 +1,242 @@
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
+
+import std/options, stew/results, chronicles, chronos, metrics, bearssl/rand
+import
+  ../node/peer_manager,
+  ../utils/requests,
+  ./protocol_metrics,
+  ./common,
+  ./rpc,
+  ./rpc_codec
+
+when defined(waku_exp_store_resume):
+  import std/[sequtils, times]
+  import ../waku_archive
+  import ../waku_core/message/digest
+
+logScope:
+  topics = "waku legacy store client"
+
+const DefaultPageSize*: uint = 20
+  # A recommended default number of waku messages per page
+
+type WakuStoreClient* = ref object
+  peerManager: PeerManager
+  rng: ref rand.HmacDrbgContext
+
+  # TODO: Move outside of the client
+  when defined(waku_exp_store_resume):
+    store: ArchiveDriver
+
+proc new*(
+    T: type WakuStoreClient, peerManager: PeerManager, rng: ref rand.HmacDrbgContext
+): T =
+  WakuStoreClient(peerManager: peerManager, rng: rng)
+
+proc sendHistoryQueryRPC(
+    w: WakuStoreClient, req: HistoryQuery, peer: RemotePeerInfo
+): Future[HistoryResult] {.async, gcsafe.} =
+  let connOpt = await w.peerManager.dialPeer(peer, WakuStoreCodec)
```

Comment:
Maybe sth to check in the future but I can't see why dialing on every query. I think it would be better to separate the `dialIn` from `query` operations :)

Reply:
What would you envision as an alternative? Dialing for each request makes this part of the code stateless which simplify things.

Do you mean separate for code organization purpose?